docs: router 2.x info (#6763)
apollographql/router: docs/source/routing/about-v2.mdx

GraphOS Router v2.x is an **Active** release with latest version {products.router.version("lts_2_latest").version}.

| Current Status | Release Date | Latest Minor | Active Date    | Maintenance Date | End of Life Date |
| -------------- | ------------ | ------------ | -------------- | ---------------- | ---------------- |
| Active    | July 2024  | v{products.router.version("lts_2_latest").version}  |  February 2025 | -     | -     |

As an [active](/graphos/reference/router-release-lifecycle#active) release:

- Backwards-compatible minor releases are planned to ship monthly.
- Minor releases can contain new features, bug fixes, and security patches.
- Breaking changes may be required in extenuating circumstances.
- [Support](https://www.apollographql.com/enterprise/support) is provided via paid support plans until end-of-life. 

## What's new in router v2.x

Below are the key highlights of the new and updated capabilities in router v2.x.

> For detailed descriptions of updates in router v2.x, go to the [router 1.x to 2.x upgrade guide](/graphos/reference/upgrade/from-router-v1). 

### Simplified integration and API orchestration of REST services using Apollo Connectors

> The router introduces **Apollo Connectors for REST** to enable the router to directly call REST APIs alongside traditional GraphQL subgraphs. It improves the interoperability of the router with existing API architectures without the need to write tedious API integration code.

Reviewer: ```suggestion
Apollo Router 2.0 introduces Apollo Connectors for REST, a declarative way to integrate REST services into supergraphs. By defining REST API integrations in GraphQL schema, you can orchestrate calls across your fleet of APIs and other services to produce a single response from a federated GraphQL API.
```

I received this edit from the Connectors team for my What's New doc. Applying it here

Author: ```suggestion
The router v2.x introduces Apollo Connectors for REST, a declarative way to integrate REST services into supergraphs. By defining REST API integrations in GraphQL schema, you can orchestrate calls across your fleet of APIs and other services to produce a single response from a federated GraphQL API.
```

### Improved query planner performance

The router is powered by a new **Native Query Planner** that achieves performance at scale with demonstrable improvement in CPU and memory utilization. It's written in Rust and replaces the legacy query planner written in JavaScript, thereby utilizing a simplified architecture and ensuring more efficient and performant query execution.

> Learn more about the [native query planner](graphos/routing/query-planning/native-query-planner).

### Predictable resource utilization and availability with back-pressure management

The router introduces **back-pressure management** that enables more predictable memory and CPU consumption. It's especially beneficial for Kubernetes deployments, where the router can now provide timely signals to trigger Horizontal Pod Autoscaling (HPA) and maintain availability. 

### Secure and stable ecosystem

The router depends on various external libraries, or crates, in the Rust ecosystem for networking, JSON processing, error handling, and more. With router v2.x, these dependencies have been updated to provide the most secure, stable, and performant runtime platform. 

> Updated crates include `axum`, `http`, `hyper`, `opentelemetry`, and `redis`. Note that some updates may cause breaking changes.

Reviewer: Worth adding a note reminding the reader to refer to the upgrade guide for more details?
```suggestion
Updated crates include `axum`, `http`, `hyper`, `opentelemetry`, and `redis`. Note that some updates may cause breaking changes, please see [upgrade guide](https://www.apollographql.com/docs/graphos/reference/migration/from-router-v1) for more details.
```
